src/lib.rs

```diff
@@ -1,8 +1,530 @@
-#[macro_use] extern crate log;
+#[cfg(feature = "cli")] use std::str::FromStr;
+
+use heck::ToUpperCamelCase;
+use k8s_openapi::apiextensions_apiserver::pkg::apis::apiextensions::v1::{
+    CustomResourceDefinition, CustomResourceDefinitionVersion,
+};
+use kube::{core::Version, ResourceExt};
 
 mod analyzer;
-pub use analyzer::{analyze, Config};
-mod output;
-pub use output::{format_docstr, Container, MapType, Member, Output};
+
 mod derive;
-pub use derive::Derive;
+mod output;
+
+pub use self::{
+    analyzer::{analyze, Config},
+    derive::Derive,
+    output::{format_docstr, Container, MapType, Member, Output},
+};
+
+/// Supported values for `kube`'s [`schema`] attribute.
+///
+/// [schema]: https://docs.rs/kube/latest/kube/derive.CustomResource.html#kubeschema--mode
+#[derive(
+    // std
+    Eq,
+    Copy,
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    // strum
+    strum::Display,
+    strum::AsRefStr,
+)]
+#[cfg_attr(feature = "cli", derive(strum::EnumString))]
+#[strum(ascii_case_insensitive, serialize_all = "lowercase")]
+pub enum SchemaMode {
+    /// Instruct `kube` to expect `JsonSchema` to be implemented manually for `kopium` generated type(s).
+    Manual,
+
+    /// Instruct `kube` to automatically derive a `JsonSchema` implementation for `kopium` generated type(s).
+    Derived,
+
+    /// Instruct `kube` to skip deriving a `JsonSchema` implementation for `kopium` generated type(s) entirely.
+    ///
+    /// **NOTE**: the resulting CRD cannot be applied to a cluster without manual fiddling to add an OpenAPI schema.
+    #[default]
+    Disabled,
```

Comment:
ah, you are picking Disabled as a default here because we already have the canonical schema.

i think this may make sense.
maybe a (later) future addition could be to attach the canonical schema in Manual mode.

Reply:
Correct, and yeah - that seems like a fantastic idea actually